spec.html

```diff
@@ -34672,6 +34672,56 @@ <h1>Static Semantics: Early Errors</h1>
         </ul>
       </emu-clause>
 
+      <emu-clause id="sec-thenumberofleftcapturingparensin" type="abstract operation">
+        <h1>
+          Static Semantics: TheNumberOfLeftCapturingParensIn (
+            _node_: a Parse Node,
+          ): a non-negative integer
+        </h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>It returns the number of left-capturing parentheses in _node_. A left-capturing parenthesis is any `(` pattern character that is matched by the `(` terminal of the <emu-grammar>Atom :: `(` GroupSpecifier Disjunction `)`</emu-grammar> production.</dd>
+        </dl>
+        <emu-alg>
+          1. Assert: _node_ is an instance of a production in <emu-xref href="#sec-patterns">the RegExp Pattern grammar</emu-xref>.
+          1. Return the number of <emu-grammar>Atom :: `(` GroupSpecifier Disjunction `)`</emu-grammar> Parse Nodes enclosed by _node_.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-thenumberofleftcapturingparensinthepatterncontaining" type="abstract operation">
+        <h1>
+          Static Semantics: TheNumberOfLeftCapturingParensInThePatternContaining (
+            _node_: a Parse Node,
+          ): a non-negative integer
+        </h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>It returns the number of left-capturing parentheses in the pattern that encloses _node_.</dd>
+        </dl>
+        <emu-alg>
+          1. Assert: _node_ is an instance of a production in <emu-xref href="#sec-patterns">the RegExp Pattern grammar</emu-xref>.
+          1. Let _pattern_ be the |Pattern| that contains _node_.
+          1. Return TheNumberOfLeftCapturingParensIn(_pattern_).
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-thenumberofleftcapturingparenstotheleftof" type="abstract operation">
+        <h1>
+          Static Semantics: TheNumberOfLeftCapturingParensToTheLeftOf (
+            _node_: a Parse Node,
+          ): a non-negative integer
+        </h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>It returns the number of left-capturing parentheses within the enclosing pattern that occur to the left of _node_.</dd>
+        </dl>
+        <emu-alg>
+          1. Assert: _node_ is an instance of a production in <emu-xref href="#sec-patterns">the RegExp Pattern grammar</emu-xref>.
+          1. Let _pattern_ be the |Pattern| that contains _node_.
+          1. Return the number of <emu-grammar>Atom :: `(` GroupSpecifier Disjunction `)`</emu-grammar> Parse Nodes enclosed by _pattern_ that occur prior to or enclosing _node_.
```

Comment:
```suggestion
          1. Return the number of <emu-grammar>Atom :: `(` GroupSpecifier Disjunction `)`</emu-grammar> Parse Nodes enclosed by _pattern_ that occur prior to or enclose _node_.
```

as a matter of English grammar, though also for consistency it should maybe be s/enclose/contain/.

Reply:
Oh, I didn't think `that occur [...] enclosing _node_` was an intended reading. That would be an odd construction.

Anyway,  current wording is fine by me. I guess "to the left of" is maybe a little bit more consistent than "before", if you feel like changing it, but I think it's fine either way.